### PR TITLE
US662948: Revisiting reusable annotation

### PR DESCRIPTION
**Pull request by nikhilahuja**
## Description
Revisiting shared entities

## Checklist
- [x] Pull Request Created
- [ ] Unit tests have been added
- [ ] Integration/Functional test cases have been written and automated

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java [hunk 1]
```diff
@@ -58,18 +58,12 @@ public void addDependentBundle(DependentBundle dependentBundle){
...
         }
-
-        StringBuilder namespace = new StringBuilder();
-        if (StringUtils.isNotBlank(getProjectInfo().getGroupName())) {
-            namespace.append(getProjectInfo().getGroupName());
-            namespace.append(".");
-        }
-        namespace.append(getAnnotatedBundleName());
-        return namespace.toString();
+        return getProjectInfo().getGroupName() + '.' + getAnnotatedBundleName();
```
Comment (uppoju) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java:66]: when group name is empty we should add "."

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java [hunk 2]
```diff
@@ -135,11 +130,12 @@ private BundleMetadata buildFullBundleMetadata(final List<Entity> entities, fina
 
     private Collection<Metadata> getDefinedEntitiesMetadata(final List<Entity> definedEntities) {
         return definedEntities.stream().filter(FILTER_NON_ENV_ENTITIES_EXCLUDING_FOLDER)
-                .map(e -> ((GatewayEntity)e.getGatewayEntity()).getMetadata()).collect(Collectors.toList());
+                .map(e -> (e.getGatewayEntity()).getMetadata()).collect(Collectors.toList());
     }
 
-    private boolean isBundleContainsReusableEntity (final AnnotatedBundle annotatedBundle) {
-        return annotatedBundle.getAnnotatedEntity().isReusable();
+    private boolean isBundleContainsSharedEntity(final AnnotatedBundle annotatedBundle) {
```
Comment (uppoju) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java:136]: why do we have to check individual policy or encass, if annotation is there then it is shared right

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java [hunk 3]
```diff
@@ -80,7 +73,7 @@ public Integer getOrder() {
         return ORDER;
     }
 
-    private String getPolicyId(String policyWithPath, Bundle bundle, AnnotatedBundle annotatedBundle) {
+    private String getPolicyId(String policyWithPath, Bundle bundle) {
```
Comment (uppoju) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java:76]: dependencies should be taken from main bundle file, if we are passing AnnotatedBundle then dependencies should be taken from main bundle present in the AnnotatedBundle

**Comment by nikhilahuja** [comment]
Closing this PR because of merge conflicts. Created new PR #283

**Pull request closed without merging**